### ISSUE TITLE
Validation fails on devfile 2.0.0 with unknown ephemeral field on volume

### ISSUE THREAD

**Issue by feloy** [devfile/api#697]
/kind bug

### Which area this bug is related to?

/area library

## What versions of software are you using?

**Operating System:**

Fedora Linux

**Go Pkg Version:**

## Bug Summary
**Describe the bug:**

When saving a devfile, the devfile library is adding an ephemeral field to the volume structure on devfile v2.0, but this field is supported starting at v2.1 only. 

**To Reproduce:**

```
$ odo create nodejs
$ odo storage create vol1
$ odo describe
# works
$ odo storage create vol2
$ odo describe
 ✗  invalid devfile schema. errors :
- components.1.volume: Additional property ephemeral is not allowed
```

**Suggestion on how to fix the bug**

The problem seems very similar to https://github.com/devfile/api/issues/675

**Comment by feloy** [comment]
CC @anandrkskd  for PR https://github.com/openshift/odo/pull/5252

**Comment by yangcao77** [comment]
This is the boolean value omitempty issue which was resolved by @kim-tsao  in 2.2.0. Just need to port over the change from 2.2.0 to 2.1.0 as well. This won't affect any devfile usage, but will affect the api spec. assigning to Kim to take a look.
